### PR TITLE
[SPARK-33991][CORE][WEBUI] Repair enumeration conversion error for AllJobsPage

### PR DESCRIPTION
**Pull request by yikf**
### What changes were proposed in this pull request?
For `AllJobsPage `class, `AllJobsPage` gets the schedulingMode of enumerated type by loading the `spark.scheduler.mode `configuration from Sparkconf, but an enumeration conversion error occurs when I set the value of this configuration to lowercase.

The reason for this problem is that the value of the `SchedulingMode `enumeration class is uppercase, which occurs when I configure `spark.scheduler.mode` to be lowercase.

I saw that the `#org.apache.spark.scheduler.TaskSchedulerImpl` class convert the `spark. scheduler.mode` value to uppercase, so I think it should be converted in `AllJobsPage `as well.

### Why are the changes needed?
An enumerated conversion error occurred with Spark when I set the value of this configuration to lowercase.

### How was this patch tested?
Existing tests.

### CODE REVIEW

**Comment by FelixYik** [comment]
@cloud-fan @zsxwing @jiangxb1987 @jerryshao please review my separated PR. Thanks!

**Comment by FelixYik** [comment]
> Can one of the admins verify this patch?

yes.

**Comment by yikf** [comment]
yes, please.










At 2021-01-04 22:14:02, "UCB AMPLab" <notifications@github.com> wrote:

Can one of the admins verify this patch?

**Comment by yikf** [comment]
@wangyum would you take a look of the PR, the update is very simple. Thanks very much!

**Comment by cloud-fan** [comment]
ok to test

**Comment by cloud-fan** [comment]
cc @gengliangwang

**Comment by maropu** [comment]
also cc: @sarutak

File: core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import javax.servlet.http.HttpServletRequest
+
+import org.mockito.Mockito._
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.status.AppStatusStore
+import org.apache.spark.status.api.v1
+
+class AllJobsPageSuite extends SparkFunSuite {
```
Comment (sarutak) [inline core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala:28]: Is it sufficient to add a test to `UISeleniumSuite` rather than adding a new Suite?

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala [hunk 2]
```diff
@@ -279,13 +289,14 @@ private[ui] class AllJobsPage(parent: JobsTab, store: AppStatusStore) extends We
 
     val schedulingMode = store.environmentInfo().sparkProperties.toMap
```
Comment (sarutak) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala:290]: Could you replace this duplicated code with the method call of `getSchedulingMode()`, or if we can avoid adding the new Suite as I mentioned elsewhere, we can remove `getSchedulingMode`.

Comment (yikf) [reply core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala:290]: Hi @sarutak ,Thanks for reviewin，I see your thoughts, I will add the test to `UISeleniumSuite` and  replace this duplicated code with the method call of `getSchedulingMode()`.

File: core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala (hunk 1 above)
Comment (yikf) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: Hi @sarutak I found that there are tests for method `render`,So we can remove `getSchedulingMode`,please review, Thanks very much!

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala [hunk 3]
```diff
@@ -277,15 +277,17 @@ private[ui] class AllJobsPage(parent: JobsTab, store: AppStatusStore) extends We
...
       .get(SCHEDULER_MODE.key)
-      .map { mode => SchedulingMode.withName(mode).toString }
+      .map { mode => SchedulingMode.withName(mode.toUpperCase(Locale.ROOT)).toString }
       .getOrElse("Unknown")
 
     val summary: NodeSeq =
       <div>
         <ul class="list-unstyled">
           <li>
+
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala:290]: nit: unnecessary change

Comment (yikf) [reply core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala:290]: But an enumerated conversion error occurred with Spark when I set the value of this spark.scheduler.mode to 'fair'.

**Comment by yikf** [comment]
@sarutak why fails SparkR unit tests， file:/tmp/RtmpIMrsYG/sparkr-test2375f14226616.parquet/_temporary does not exist？

**Comment by sarutak** [comment]
@yikf The R test failure irrelevant to this change.

File: core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala (hunk 1 above)
Comment (sarutak) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: The latest change doesn't include test. Could you add one to `UISeleniumSuite`?

**Comment by yikf** [comment]
> @yikf The R test failure irrelevant to this change.

@sarutak Ok, please review... Thanks very much!

File: core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala (hunk 1 above)
Comment (yikf) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: > The latest change doesn't include test. Could you add one to `UISeleniumSuite`?

oh, Because i found that there are tests for method `render` yestoday.

Comment (sarutak) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: I mean we need to test whether the page is rendered and the scheduling mode is displayed with no error when we visit `http://localhost:4040/jobs/` even though we configure the scheduling mode like `spark.scheduler.mode=fair` instead of `spark.scheculer.mode=FAIR`.
Currently, we can get the following error when we visit `http://localhost:4040/jobs/` if we set the scheduling mode in lower case.
scheduling_mode_in_lower_case

Comment (yikf) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: In `UISeleniumSuite.scala`，due to import scalaTest，I may not be able to mock，and I see that `AllJobsPage.scala`  doesn't has own suite，May i add a new suite？thx

Comment (yikf) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: Hi @sarutak ，I've added the test in `UISeleniumSuite`, thx.

Comment (sarutak) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: What I intend is adding a test using Selenium like other tests in `UISeleniumSuite`., then you don't need to add `getScheduleMode`.
Here is an example.
https://github.com/apache/spark/blob/71d261ab8fb3e7fb22d2687b8e038129ca766a65/core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala#L723-738

Comment (yikf) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: Hi @sarutak , I don't understand that can i get <strong> element?
`val conf = Map("spark.scheduler.mode" -> "fair")
    withSpark(newSparkContext(additionalConfs = conf)) { sc =>
      val rdd = sc.parallelize(0 to 100, 100).repartition(10).cache()
      rdd.count()

      eventually(timeout(5.seconds), interval(50.milliseconds)) {
        goToUi(sc, "/jobs")
        find(linkText("Scheduling Mode")).get.text should contain("FAIR")
      }
    }`

Comment (yikf) [reply core/src/test/scala/org/apache/spark/ui/jobs/AllJobsPageSuite.scala]: Hi @sarutak , I am so sorry，I haven't created PR before，I understand what you mean now。

I have removed `getSchedulingMode` and add a suit like other tests in `UISeleniumSuite`,Thanks very much!

**Review by srowen — COMMENTED** [review]
The change looks OK to me.

**Comment by yikf** [comment]
> The change looks OK to me.

Hi，Thank  you for your response，Can you merge it into master branch for me? Thank you so much!

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master
